src/main/java/com/mwt/explorers/EpsilonGreedyExplorer.java

```diff
@@ -45,22 +45,23 @@ public void updatePolicy(Policy<T> newPolicy) {
     this.defaultPolicy = newPolicy;
   }
 
-  public ChosenAction chooseAction(long saltedSeed, T context) {
+  public DecisionTuple chooseAction(long saltedSeed, T context) {
+    int numActionsForContext = getNumActions(context);
     Random random = new Random(saltedSeed);
```

Comment:
We definitely want black-box compatibility.  In my (painful) experience 
this means that a common implementation of hashing is unavoidable.

-John

On 06/18/2015 11:59 PM, Markus Cozowicz wrote:
>
> In src/main/java/com/mwt/explorers/EpsilonGreedyExplorer.java 
> <https://github.com/multiworldtesting/explore-java/pull/2#discussion_r32800057>:
>
> > @@ -45,22 +45,23 @@ public void updatePolicy(Policy<T> newPolicy) {
> >      this.defaultPolicy = newPolicy;
> >    }
> >
> > -  public ChosenAction chooseAction(long saltedSeed, T context) {
> > +  public DecisionTuple chooseAction(long saltedSeed, T context) {
> > +    int numActionsForContext = getNumActions(context);
> >      Random random = new Random(saltedSeed);
>
> @lhoang29 <https://github.com/lhoang29> are we going to use the 
> built-in C# Random generator as well or re-implement the current C 
> version?
> If we want blackbox test compat, we might all have to settle for the 
> same random generator
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/multiworldtesting/explore-java/pull/2/files#r32800057>.
>